src/main/java/com/google/codeu/servlets/MarkerMessagesServlet.java

```diff
@@ -87,8 +74,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     double score = sentiment.getScore();
     languageService.close();
 
-    Message message = new Message(user, userText, score, parentKey);
-    datastore.storeMessage(message);
+    Message message = new Message(user, userText, score);
+    System.out.println(parentKey + parentKind);
+    if (parentKey != null && parentKind != null)
+      datastore.storeMessage(message, KeyFactory.createKey(parentKind, parentKey));
```

Comment:
I think you can pass string representations of keys around: https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Java_Generating_keys

Something like:
```
String personKeyStr = KeyFactory.keyToString(k);

// Some time later (for example, after using personKeyStr in a link).
Key personKey = KeyFactory.stringToKey(personKeyStr);
Entity person = datastore.get(personKey);
```

This way also works but its a little fragile and you have to pass around two strings, rather than just the actual key itself.

Reply:
Implementation updated